File: Assets/My Assets/Scripts/Gameplay/Player/PlayerManager.js
﻿//Manages all of our basic game variables and UI.
var minerals : int = 0; //How much money do we got, homie?
var player : GameObject; //Who are we looking at, cuz?
var mineralsUI : UI.Text;
var controlsUI : GameObject;
var aiUI : UI.Text;
var myAI : AIMode = 0;
function Start(){
	if(PlayerPrefs.HasKey("Minerals") == false){ //Yo, you got that written down dawg?
		PlayerPrefs.SetInt("Minerals",0);
		PlayerPrefs.Save(); //Ya boy, that's how we do.
	}
	else{ //What it say tho
		minerals = PlayerPrefs.GetInt("Minerals"); //I hear you homie
	}
}
function setMineralUI(){
	mineralsUI.text = "Minerals: "+minerals;
}
function setAIUI(){ //Set our AI State UI.
	var myText = "AI: ";
	if(myAI == 0){
		myText+="Sentry";
	}
	else{
		if(myAI == 1){
			myText+="Follow";
		}
		else{
			if(myAI == 2){
				myText+="Drift";
			}
		}
	}
	aiUI.text = myText;
}
function Update(){
	setMineralUI();
	if(Input.GetKeyDown("/")){
		Application.Quit();
	}
	if(Input.GetButtonDown("AIMode")){
		myAI+=1;
		if(myAI > 2){
			myAI = 0;
		}
		setAIUI();
	}
	if(Input.GetButtonDown("Controls")){
		if(controlsUI.active == true){
			controlsUI.active = false;
		}
		else{
			controlsUI.active = true;
		}
	}
}

File: Assets/My Assets/Scripts/Gameplay/Player/PlayerManager.cs
using UnityEngine;
using UnityEngine.UI;
using System.Collections;

[System.Serializable]
public class PlayerManager : MonoBehaviour
{
    //Manages all of our basic game variables and UI.
    public int minerals; //How much money do we got, homie?
    public GameObject player; //Who are we looking at, cuz?
    public Text mineralsUI;
    public GameObject controlsUI;
    public Text aiUI;
    public AIMode myAI;
    public Text squadUI;
    //var myOptions :GameObject;
    public virtual void Start()
    {
        if (PlayerPrefs.HasKey("Minerals") == false) //Yo, you got that written down dawg?
        {
            PlayerPrefs.SetInt("Minerals", 0);
            PlayerPrefs.Save(); //Ya boy, that's how we do.
        }
        else
        {
             //What it say tho
            this.minerals = PlayerPrefs.GetInt("Minerals"); //I hear you homie
        }
        this.setAIUI();
    }

    public virtual void setMineralUI()
    {
        this.mineralsUI.text = "Minerals: " + this.minerals;
    }

    public virtual void setAIUI() //Set our AI State UI.
    {
        string myText = "Fleet Orders: ";
        if (this.myAI == (AIMode) 0)
        {
            myText = myText + "Sentry";
        }
        else
        {
            if (this.myAI == (AIMode) 1)
            {
                myText = myText + "Follow";
            }
            else
            {
                if (this.myAI == (AIMode) 2)
                {
                    myText = myText + "Drift";
                }
            }
        }
        this.aiUI.text = myText;
    }

    public virtual void Update()
    {
        this.setMineralUI();
        if (Input.GetButtonDown("Menu") && (GameObject.FindGameObjectWithTag("PauseMenu") == null))
        {
            AsyncOperation @async = Application.LoadLevelAdditiveAsync("options");
            //yield async;
            Debug.Log("Loading complete");
            Time.timeScale = 0;
        }
        if (Input.GetButtonDown("AIMode"))
        {
            this.myAI = this.myAI + 1;
            if (this.myAI > (AIMode) 2)
            {
                this.myAI = (AIMode) 0;
            }
            this.setAIUI();
        }
        if (Input.GetButtonDown("Controls"))
        {
            if (this.controlsUI.active == true)
            {
                this.controlsUI.active = false;
                PlayerPrefs.SetInt("Controls", 0);
            }
            else
            {
                this.controlsUI.active = true;
                PlayerPrefs.SetInt("Controls", 1);
            }
            PlayerPrefs.Save();
        }
    }

}